examples/ErrorHandling/HandleRateExceededError.php

```diff
@@ -0,0 +1,263 @@
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int $adGroupId the ad group ID to validate keywords from
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId
+    ) {
+        // Sequentially sends the requests.
+        for ($i = 0; $i < self::NUM_REQUESTS; $i++) {
+            // Creates operations.
+            $operations = self::createAdGroupCriterionOperations($customerId, $adGroupId, $i);
+
+            try {
+                $retryCount = 0;
+                $retrySeconds = self::RETRY_SECONDS;
+                while ($retryCount < self::NUM_RETRIES) {
+                    try {
+                        // Sends request.
+                        self::requestMutateAndDisplayResult(
+                            $googleAdsClient,
+                            $customerId,
+                            $operations
+                        );
+                        break;
+                    } catch (GoogleAdsException $googleAdsException) {
+                        foreach ($googleAdsException->getGoogleAdsFailure()
+                                     ->getErrors() as $googleAdsError) {
+                            // Checks if any of the errors are QuotaError.RESOURCE_EXHAUSTED or
+                            // QuotaError.RESOURCE_TEMPORARILY_EXHAUSTED.
+                            if ($googleAdsError->getErrorCode()->getQuotaError() ==
+                                QuotaError::RESOURCE_EXHAUSTED ||
+                                $googleAdsError->getErrorCode()->getQuotaError() ==
+                                QuotaError::RESOURCE_TEMPORARILY_EXHAUSTED) {
+                                printf(
+                                    "Received rate exceeded error, retry after %d seconds.%s",
+                                    $retrySeconds,
+                                    PHP_EOL
+                                );
+                                sleep($retrySeconds);
+                                $retryCount++;
+                                // Uses an exponential back-off policy.
+                                $retrySeconds *= 2;
+                                break;
+                            }
+                        }
+                        // Bubbles up when not a RateExceededError
+                        throw $googleAdsException;
```

Comment:
Should this be in an "else" of the if clause that checks quota error above?
If not, it will be thrown right away after you break from the above loop.

Reply:
The GoogleAdsException may contain multiple errors so I bubble up after the loop to make sure all of them have been tested.

Reply:
Good point, I add a flag to only bubble up when there is no RateExceededError found in any of the returned errors.